pkg/scheduler/apis/config/v1beta1/defaults.go

```diff
@@ -205,6 +205,13 @@ func SetDefaults_VolumeBindingArgs(obj *v1beta1.VolumeBindingArgs) {
 	}
 }
 
+func SetDefaults_NodeResourcesBalancedAllocationArgs(obj *v1beta1.NodeResourcesBalancedAllocationArgs) {
+	if len(obj.Resources) == 0 {
+		// If no resources specified, used the default set.
+		obj.Resources = append(obj.Resources, defaultResourceSpec...)
+	}
```

Comment:
It should also add weight=1, if `obj.Resources` is not empty.

Reply:
Can we hit this situation for both default setting is needed while the `obj.Resources` is not empty?

https://github.com/kubernetes/kubernetes/blob/e8cf412e5ec70081477ad6f126c7f7ef7449109c/pkg/scheduler/framework/runtime/framework.go#L416-L419

Reply:
you could have someone adding:

```yaml
resources:
- name: cpu
- name: memory
- name: gpu
```

In fact, that's what they should do, as weight is meaningless. But I prefer we add a type that doesn't have weight.

Reply:
What I meant was that if you have a config like below,
```yaml
  - args:
      apiVersion: kubescheduler.config.k8s.io/v1beta1
      resources:
      - name: hugepages-resource1
      - name: cpu
      - name: memory
    name: NodeResourcesBalancedAllocation
```

It will be captured by the code here, and the `decode` to set the defaults will not got called. https://github.com/kubernetes/kubernetes/blob/e8cf412e5ec70081477ad6f126c7f7ef7449109c/pkg/scheduler/framework/runtime/framework.go#L416-L422

and yes, I will create another type without weight at all.

Reply:
Done